src/utilities/CSX.ts

```diff
@@ -0,0 +1,18 @@
+type ClassListDefinition = Record<string, boolean>;
+
+export const CSX = (classListDefinition: ClassListDefinition): string => {
+    const classNames = Object.entries(classListDefinition)
+        .reduce<string[]>(
+            (classList, [className, condition]) => {
+                if (condition) {
+                    classList.push(className);
+                }
+
+                return classList;
+            },
+            []
+        )
+        .join(" ");
+
+    return classNames;
+};
```

Comment:
Would you be able to explain what this does?

Reply:
It takes an object where the keys are the class name and the values are booleans indicating whether or not the class name should be applied. 

The method converts the objects to an array, converts it into an array of strings (the class names that should be applied) and then joins them with a space between them since the class attribute expects class names to be separated via white space

Reply:
Is the goal to get a list of applied class names? How would you feel about changing the function name to reflect that?

Reply:
I have no love for the function name `CSX` I would just prefer something on the shorter side so as to not clutter up the JSX too much